src/test/java/com/amazonaws/encryptionsdk/keyrings/AwsKmsSymmetricRegionDiscoveryKeyringTest.java

```diff
@@ -0,0 +1,309 @@
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class AwsKmsSymmetricRegionDiscoveryKeyringTest {
+
+    private static final CryptoAlgorithm ALGORITHM_SUITE = CryptoAlgorithm.ALG_AES_256_GCM_IV12_TAG16_HKDF_SHA256;
+    private static final SecretKey PLAINTEXT_DATA_KEY = new SecretKeySpec(generate(ALGORITHM_SUITE.getDataKeyLength()), ALGORITHM_SUITE.getDataKeyAlgo());
+    private static final Map<String, String> ENCRYPTION_CONTEXT = Collections.singletonMap("myKey", "myValue");
+    private static final String KEY_NAME_1 = "arn:aws:kms:us-east-1:999999999999:key/key1-23bv-sdfs-werw-234323nfdsf";
+    private static final String KEY_NAME_2 = "arn:aws:kms:us-east-1:999999999999:key/key2-02ds-wvjs-aswe-a4923489273";
+    private static final String DIFFERENT_REGION_KEY_NAME = "arn:aws:kms:us-west-2:999999999999:key/key2-02ds-wvjs-aswe-a4923489273";
+    private static final String DIFFERENT_AWS_ACCOUNT_ID_KEY_NAME = "arn:aws:kms:us-east-1:000000000000:key/key2-02ds-wvjs-aswe-a4923489273";
+    private static final String AWS_ACCOUNT_ID = "999999999999";
+    private static final String AWS_REGION = "us-east-1";
+    private static final EncryptedDataKey ENCRYPTED_KEY_1 = new KeyBlob(AWS_KMS_PROVIDER_ID,
+        KEY_NAME_1.getBytes(PROVIDER_ENCODING), generate(ALGORITHM_SUITE.getDataKeyLength()));
+    private static final EncryptedDataKey ENCRYPTED_KEY_2 = new KeyBlob(AWS_KMS_PROVIDER_ID,
+        KEY_NAME_2.getBytes(PROVIDER_ENCODING), generate(ALGORITHM_SUITE.getDataKeyLength()));
+    private static final EncryptedDataKey ENCRYPTED_DIFFERENT_REGION_KEY = new KeyBlob(AWS_KMS_PROVIDER_ID,
+        DIFFERENT_REGION_KEY_NAME.getBytes(PROVIDER_ENCODING), generate(ALGORITHM_SUITE.getDataKeyLength()));
+    private static final EncryptedDataKey ENCRYPTED_DIFFERENT_AWS_ACCOUNT_ID_KEY = new KeyBlob(AWS_KMS_PROVIDER_ID,
+        DIFFERENT_AWS_ACCOUNT_ID_KEY_NAME.getBytes(PROVIDER_ENCODING), generate(ALGORITHM_SUITE.getDataKeyLength()));
+    @Mock(lenient = true)
+    private DataKeyEncryptionDao dataKeyEncryptionDao;
+    private Keyring keyring;
+
+    @BeforeEach
+    void setup() {
+        when(dataKeyEncryptionDao.decryptDataKey(ENCRYPTED_KEY_1, ALGORITHM_SUITE, ENCRYPTION_CONTEXT))
+            .thenReturn(new DecryptDataKeyResult(KEY_NAME_1, PLAINTEXT_DATA_KEY));
+        when(dataKeyEncryptionDao.decryptDataKey(ENCRYPTED_KEY_2, ALGORITHM_SUITE, ENCRYPTION_CONTEXT))
+            .thenReturn(new DecryptDataKeyResult(KEY_NAME_2, PLAINTEXT_DATA_KEY));
+        when(dataKeyEncryptionDao.decryptDataKey(ENCRYPTED_DIFFERENT_AWS_ACCOUNT_ID_KEY, ALGORITHM_SUITE, ENCRYPTION_CONTEXT))
+            .thenReturn(new DecryptDataKeyResult(DIFFERENT_AWS_ACCOUNT_ID_KEY_NAME, PLAINTEXT_DATA_KEY));
+        keyring = new AwsKmsSymmetricRegionDiscoveryKeyring(dataKeyEncryptionDao, AWS_REGION, AWS_ACCOUNT_ID);
+    }
+
+    @Test
+    void testMalformedArns() {
```

Comment:
Similar to other comments and throughout this file, why test with two encrypted data keys such that the second one will always work?

Reply:
I believe I've already updated this - kept 1 "fail then succeed to decrypt" case and made all other failure cases exclusively failures.